### PR TITLE
Feat/missing tx events

### PR DESCRIPTION
**Pull request by cbetta**
Adds a lot of the missing TX event as well as the generic event type for use with SDKs.

### CODE REVIEW

File: reference/openapi.yaml [hunk 1]
```diff
@@ -771,8 +771,14 @@ components:
       $ref: "./resources/actions.yaml"
     Address:
       $ref: "./resources/address.yaml"
-    AntiFraudDecisionTransactionEvent:
+    AntiFraudDecisionEvent:
```
Comment (cbetta) [inline reference/openapi.yaml:774]: [Info] Renamed to match the event name

File: reference/openapi.yaml [hunk 2]
```diff
@@ -1039,14 +1057,16 @@ components:
       $ref: "./resources/three_d_secure_data_v1_v2.yaml"
     ThreeDSecureError:
       $ref: "./resources/three_d_secure_error.yaml"
-    ThreeDSecureSuccessTransactionEvent:
+    ThreeDSecureSuccessEvent:
```
Comment (cbetta) [inline reference/openapi.yaml:1060]: [Info] Renamed to match the event name

File: reference/openapi.yaml [hunk 3]
```diff
@@ -1039,14 +1057,16 @@ components:
...
       $ref: "./resources/transaction_history_events/three_d_secure_success.yaml"
     ThreeDSecureSummary:
       $ref: "./resources/three_d_secure_summary.yaml"
     TokenizedRequest:
       $ref: "./request-bodies/tokenized_request.yaml"
     Transaction:
       $ref: "./resources/transaction.yaml"
+    TransactionHistoryEvent:
+      $ref: "./resources/transaction_history_event.yaml"
```
Comment (cbetta) [inline reference/openapi.yaml:1069]: [Info] Generic event so we could even use this in the SDKs

File: reference/resources/transaction_history_events.yaml [hunk 4]
```diff
@@ -14,8 +14,17 @@ properties:
...
+        - $ref: "../openapi.yaml#/components/schemas/AntiFraudDecisionErrorEvent"
+        - $ref: "../openapi.yaml#/components/schemas/AntiFraudDecisionEvent"
+        - $ref: "../openapi.yaml#/components/schemas/AntiFraudDecisionSkippedEvent"
+        - $ref: "../openapi.yaml#/components/schemas/AntiFraudWebhookEvent"
+        - $ref: "../openapi.yaml#/components/schemas/PaymentConnectorResponseTransactionAuthorizationFailedEvent"
+        - $ref: "../openapi.yaml#/components/schemas/PaymentConnectorResponseTransactionAuthorizationSucceededEvent"
+        - $ref: "../openapi.yaml#/components/schemas/PaymentConnectorResponseTransactionCaptureDeclinedEvent"
+        - $ref: "../openapi.yaml#/components/schemas/PaymentConnectorResponseTransactionCaptureFailedEvent"
+        - $ref: "../openapi.yaml#/components/schemas/PaymentConnectorResponseTransactionCaptureSucceededEvent"
+        - $ref: "../openapi.yaml#/components/schemas/PaymentConnectorResponseTransactionDeclinedEvent"
```
Comment (cbetta) [inline reference/resources/transaction_history_events.yaml:27]: [Info] Add a lot of the missing events.

File: reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_failed.yaml [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+
+      payment_service_definition_id:
+        type: string
+        description: The payment service definition used.
+
+      payment_service_transaction_id:
+        type: string
+        description:
+          The external ID of the transaction as set by the payment service.
+        example: "transaction-1234"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_failed.yaml:61]:
```suggestion
        example: "transaction-1234"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_failed.yaml [hunk 6]
```diff
@@ -0,0 +1,162 @@
...
+          - authorization_failed
+        description: |-
+          The new status code for the transaction. This is always
+          set to `authorization_failed`.
+        example: authorization_failed
+
+      code:
+        type: string
+        description: A raw response code returned for the failure.
+        example: "INVALID DATA"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_failed.yaml:75]:
```suggestion
        example: "INVALID DATA"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_failed.yaml [hunk 7]
```diff
@@ -0,0 +1,162 @@
...
+        description: The type of instrument used for this transaction.
+
+      retry_rule:
+        type: string
+        description:
+          Whether this transaction is retried as a failure or decline.
+        enum:
+          - failure
+          - retriable_decline
+        example: failure
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_failed.yaml:97]:
```suggestion
        example: failure
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_succeeded.yaml [hunk 8]
```diff
@@ -0,0 +1,157 @@
...
+
+      payment_service_definition_id:
+        type: string
+        description: The payment service definition used.
+
+      payment_service_transaction_id:
+        type: string
+        description:
+          The external ID of the transaction as set by the payment service.
+        example: "transaction-1234"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_succeeded.yaml:61]:
```suggestion
        example: "transaction-1234"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_succeeded.yaml [hunk 9]
```diff
@@ -0,0 +1,157 @@
...
+        description: The type of instrument used for this transaction.
+
+      retry_rule:
+        type: string
+        description:
+          Whether this transaction is retried as a failure or decline.
+        enum:
+          - failure
+          - retriable_decline
+        example: failure
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_authorization_succeeded.yaml:92]:
```suggestion
        example: failure
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml [hunk 10]
```diff
@@ -0,0 +1,160 @@
...
+
+      payment_service_definition_id:
+        type: string
+        description: The payment service definition used.
+
+      payment_service_transaction_id:
+        type: string
+        description:
+          The external ID of the transaction as set by the payment service.
+        example: "transaction-1234"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml:61]:
```suggestion
        example: "transaction-1234"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml [hunk 11]
```diff
@@ -0,0 +1,160 @@
...
+        enum:
+          - processing
+          - buyer_approval_pending
+          - authorization_succeeded
+          - authorization_failed
+          - authorization_declined
+          - capture_pending
+          - capture_succeeded
+          - authorization_void_pending
+          - authorization_voided
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml:77]:
```suggestion
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml [hunk 12]
```diff
@@ -0,0 +1,160 @@
...
+
+      code:
+        type: string
+        description: A raw response code returned for the failure.
+        example: "INVALID DATA"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml:82]:
```suggestion
        example: "INVALID DATA"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml [hunk 13]
```diff
@@ -0,0 +1,160 @@
...
+        example: pan
+        enum:
+          - pan
+          - card_token
+          - redirect
+          - redirect_token
+          - googlepay
+          - applepay
+          - network_token
+        description: The type of instrument used for this transaction.
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_declined.yaml:95]:
```suggestion
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml [hunk 14]
```diff
@@ -0,0 +1,160 @@
...
+
+      payment_service_definition_id:
+        type: string
+        description: The payment service definition used.
+
+      payment_service_transaction_id:
+        type: string
+        description:
+          The external ID of the transaction as set by the payment service.
+        example: "transaction-1234"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml:61]:
```suggestion
        example: "transaction-1234"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml [hunk 15]
```diff
@@ -0,0 +1,160 @@
...
+        enum:
+          - processing
+          - buyer_approval_pending
+          - authorization_succeeded
+          - authorization_failed
+          - authorization_declined
+          - capture_pending
+          - capture_succeeded
+          - authorization_void_pending
+          - authorization_voided
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml:77]:
```suggestion
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml [hunk 16]
```diff
@@ -0,0 +1,160 @@
...
+
+      code:
+        type: string
+        description: A raw response code returned for the failure.
+        example: "INVALID DATA"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml:82]:
```suggestion
        example: "INVALID DATA"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml [hunk 17]
```diff
@@ -0,0 +1,160 @@
...
+        example: pan
+        enum:
+          - pan
+          - card_token
+          - redirect
+          - redirect_token
+          - googlepay
+          - applepay
+          - network_token
+        description: The type of instrument used for this transaction.
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_failed.yaml:95]:
```suggestion
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml [hunk 18]
```diff
@@ -0,0 +1,157 @@
...
+
+      payment_service_definition_id:
+        type: string
+        description: The payment service definition used.
+
+      payment_service_transaction_id:
+        type: string
+        description:
+          The external ID of the transaction as set by the payment service.
+        example: "transaction-1234"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml:61]:
```suggestion
        example: "transaction-1234"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml [hunk 19]
```diff
@@ -0,0 +1,157 @@
...
+        type: string
+        format: uuid
+        description: The unique ID of the payment service used.
+
+      payment_service_display_name:
+        type: string
+        description: The display name of the payment service used.
+
+      payment_service_definition_id:
+        type: string
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml:54]:
```suggestion
        type: string
        format: uuid
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml [hunk 20]
```diff
@@ -0,0 +1,157 @@
...
+        example: pan
+        enum:
+          - pan
+          - card_token
+          - redirect
+          - redirect_token
+          - googlepay
+          - applepay
+          - network_token
+        description: The type of instrument used for this transaction.
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml:83]:
```suggestion
        description: The type of instrument used for this transaction.
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml [hunk 21]
```diff
@@ -0,0 +1,157 @@
...
+
+      retry_rule:
+        type: string
+        description:
+          Whether this transaction is retried as a failure or decline.
+        enum:
+          - failure
+          - retriable_decline
+        example: failure
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_capture_succeeded.yaml:92]:
```suggestion
        example: failure
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_declined.yaml [hunk 22]
```diff
@@ -0,0 +1,140 @@
...
+    format: uuid
+
+  name:
+    type: string
+    description:
+      The name of this resource. Is always
+      `payment-connector-response-transaction-capture-declined`.
+    example: payment-connector-response-transaction-capture-declined
+    enum:
+      - payment-connector-response-transaction-capture-declined
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_declined.yaml:31]:
```suggestion
      `payment-connector-response-transaction-declined`.
    example: payment-connector-response-transaction-declined
    enum:
      - payment-connector-response-transaction-declined
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_declined.yaml [hunk 23]
```diff
@@ -0,0 +1,140 @@
...
+
+      payment_service_definition_id:
+        type: string
+        description: The payment service definition used.
+
+      payment_service_transaction_id:
+        type: string
+        description:
+          The external ID of the transaction as set by the payment service.
+        example: "transaction-1234"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_declined.yaml:61]:
```suggestion
        example: "transaction-1234"
        nullable: true
```

File: reference/resources/transaction_history_events/payment_connector_response_transaction_declined.yaml [hunk 24]
```diff
@@ -0,0 +1,140 @@
...
+          - authorization_declined
+        description: |-
+          The new status code for the transaction. This is always
+          set to `authorization_declined`.
+        example: authorization_declined
+
+      code:
+        type: string
+        description: A raw response code returned for the decline.
+        example: "INVALID DATA"
```
Comment (gryevns) [inline reference/resources/transaction_history_events/payment_connector_response_transaction_declined.yaml:75]:
```suggestion
        example: "INVALID DATA"
        nullable: true
```

**Pull request merged**